pkg/services/ngalert/api/tooling/definitions/alertmanager_validation.go

```diff
@@ -0,0 +1,65 @@
+package definitions
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/common/model"
+)
+
+// Validate normalizes a Route r, and returns errors if r is invalid.
+func (r *Route) Validate() error {
```

Comment:
All code lifted verbatim from the original `UnmarshalYAML`. No changes.

Reply:
No terribly strong opinion here.

Originally chose a method because `Validate()` is very short, and every type can have a Validate method attached. Otherwise, we'd need many methods called `ValidateX` and `ValidateY` which are longer on average.

Not a strong reason, happy to switch.

Reply:
After playing with this, I realized:

The validation code for most types in `definitions` not only validates - it validates _and normalizes_.

The signature `func ValidateRoute(r Route) error` is insufficient because the route also gets modified. It'd need to be by reference, `func ValidateRoute(r *Route) error` which is _okay_, but I'm really not a big fan of the reliance on side effects.